prog/kubernetes/kubernetes_cluster_nexus.rb

```diff
@@ -3,27 +3,38 @@
 class Prog::Kubernetes::KubernetesClusterNexus < Prog::Base
   subject_is :kubernetes_cluster
 
-  def self.assemble(name:, version:, private_subnet_id:, project_id:, location:, cp_node_count: 3, target_node_size: "standard-2", target_node_storage_size_gib: nil)
+  def self.assemble(name:, project_id:, location:, version: "v1.32", private_subnet_id: nil, cp_node_count: 3, target_node_size: "standard-2", target_node_storage_size_gib: nil)
     DB.transaction do
       unless (project = Project[project_id])
         fail "No existing project"
       end
 
-      unless ["v1.32"].include?(version)
+      unless ["v1.32", "v1.31"].include?(version)
         fail "Invalid Kubernetes Version"
       end
 
       Validation.validate_kubernetes_name(name)
       Validation.validate_kubernetes_cp_node_count(cp_node_count)
 
-      # TODO: Validate subnet location if given
-      # TODO: Validate subnet size if given
-      # TODO: Create subnet if not given
+      subnet = if private_subnet_id && private_subnet_id != ""
+        fail "Given subnet is not available in the given project" unless project.private_subnets.any? { |ps| ps.id == private_subnet_id }
+        PrivateSubnet[private_subnet_id]
```

Comment:
`project.private_subnets.find { |ps| ps.id == private_subnet_id` will return `PrivateSubnet` model.

maybe 🤔 

```suggestion
        project.private_subnets.find { _1.id == private_subnet_id } || fail "Given subnet is not available in the given project"
```

Reply:
neat!